Automattic/woocommerce-payments
# Save the generated customer id using session cookie for non logged in users

Fixes #2716 

#### Changes proposed in this Pull Request

Save the generated customer id using session cookie for non logged in users

Right now everytime a payment is executed we are creating a new customer id for every attempt a non logged in user does. Apart from being not practical, it was creating issues when attempting a failed payment if you were not logged in.

This PR solves that saving the customer id in the session cookie, so if a user is non logged in (id 0) and has an existing customer id, we can use that to continue payment intents.





#### Testing instructions

- Log out
- add items to cart and go to check out
- Use `4000000000000002` and see a card declined error
- Click on "Place Order" again without refreshing
- See the card declined error


Before this fix, after the second attempt would display this error
![Screenshot 2021-08-12 at 13 38 25](https://user-images.githubusercontent.com/2612426/129190728-216c9725-4336-478b-b0bb-ab16f04af54a.png)

-------------------
